cmd/verify.go

```diff
@@ -47,16 +48,24 @@ var debugCmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
-		var validateOpts *server.VerifySnpOpts
-		if len(teeNonce) != 0 {
-			validateOpts = &server.VerifySnpOpts{
-				Validation:   server.SevSnpDefaultValidateOpts(teeNonce),
-				Verification: &sv.Options{},
+
+		var validateOpts interface{}
+		switch attestation.GetTeeAttestation().(type) {
+		case *pb.Attestation_TdxAttestation:
+			validateOpts = &server.VerifyTdxOpts{
+				Verification: tv.DefaultOptions(),
 			}
-		} else {
-			validateOpts = &server.VerifySnpOpts{
-				Validation:   server.SevSnpDefaultValidateOpts(nonce),
-				Verification: &sv.Options{},
+		default:
```

Comment:
Added a case for sev-snp attestation.
Since verifyGceTechnology does not return an error when attestation contains GCEConfidentialTechnology_NONE, I assume that the default case should not return an error. Please correct me if it's incorrect.

Reply:
Sorry you're right. validateOpts should be nil by default.

I don't see the case handling the SNP attestation. As the code is currently written, we always try to validate with a VerifySnpOpts, even if there is no TeeAttestation in the proto.